spring-cloud-task-stream/src/main/java/org/springframework/cloud/task/batch/listener/EventEmittingChunkListener.java

```diff
@@ -0,0 +1,60 @@
+package org.springframework.cloud.task.batch.listener;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.batch.core.ChunkListener;
+import org.springframework.batch.core.ItemReadListener;
+import org.springframework.batch.core.scope.context.ChunkContext;
+import org.springframework.cloud.task.batch.listener.support.BatchJobHeaders;
+import org.springframework.cloud.task.batch.listener.support.MessagePublisher;
+import org.springframework.core.Ordered;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.util.Assert;
+
+/**
+ *  Provides informational messages around the {@link Chunk} of a batch job.
+ *
+ *  The {@link ChunkListener#beforeChunk()} and
+ *  {@link ChunkListener#afterChunk(Object)} are both no-ops in this implementation.
+ *  {@link ChunkListener#afterChunkError(Object)}.
+ *
+ * @author Ali Shahbour
+ */
+public class EventEmittingChunkListener implements ChunkListener, Ordered {
```

Comment:
My point is that I believe the original option for the chunk listener did emit messages (via the proxy) so I'd expect if we're going down this route, we'd implement that functionality.